src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -239,7 +240,9 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
                 {
                     // Create the comment file for reshelve.
                     StringBuilder comment = new StringBuilder(tfShelveset.Comment ?? string.Empty);
-                    if (!(executionContext.Variables.Build_GatedRunCI ?? true))
+                    string runCi = GetEndpointData(endpoint, Constants.EndpointData.GatedRunCI);
+                    bool? gatedRunCi = executionContext.Variables.GetBoolean(runCi);
```

Comment:
This looks like a bug. Shouldn't the value either come from the endpoint data or from the variable dictionary. The value from the endpoint data isn't a lookup key into the variable dictionary.